java/src/main/java/io/grpc/examples/wallet/AccountServer.java

```diff
@@ -97,17 +110,30 @@ private void start() throws IOException {
       Observability.registerExporters(gcpClientProject);
     }
     HealthStatusManager health = new HealthStatusManager();
+    ServerCredentials serverCredentials =
+        credentialsType == CredentialsType.XDS
+            ? XdsServerCredentials.create(InsecureServerCredentials.create())
+            : InsecureServerCredentials.create();
+    // Since the main server may be using TLS, we start a second server just for plaintext health
```

Comment:
Comments on this file apply to the other servers as well.

Don't we know if the main server is using TLS? Why not start the second server iff the main server is using TLS?

This PR has the effect of changing the behavior for all of the (non-security) scenarios described in other user guides to start up two Java servers. I don't think we want to demonstrate that as the default behavior.

Can this gate the extra health server behind a check of the credentials flag? And do we even need the health service on the secure server when running a health server on port+1?

Reply:
> Comments on this file apply to the other servers as well.
> 
> Don't we know if the main server is using TLS? Why not start the second server iff the main server is using TLS?

What you are saying makes sense but I followed the pattern from the Java xds example [here](https://github.com/grpc/grpc-java/blob/master/examples/example-xds/src/main/java/io/grpc/examples/helloworldxds/XdsHelloWorldServer.java#L73).

> 
> This PR has the effect of changing the behavior for all of the (non-security) scenarios described in other user guides to start up two Java servers. I don't think we want to demonstrate that as the default behavior.

Actually that's not true. The original server's health-service is always there so the behavior for the non-secure mode doesn't change except there is also an extra server bound to the healthPort.

> 
> Can this gate the extra health server behind a check of the credentials flag? And do we even need the health service on the secure server when running a health server on port+1?

All of these changes are possible (and one might say even make sense) but as I mentioned earlier this pattern is based on the example-xds.

Reply:
For now I haven't made any changes. Let me know what you think after taking a look at the example-xds code.

Reply:
That all makes sense. I wasn't aware of the pattern of the already existing example